lib/uri_parser.js

```diff
@@ -124,6 +124,10 @@ function parseQueryStringItemValue(key, value) {
     // deduplicate and simplify arrays
     value = value.filter((v, idx) => value.indexOf(v) === idx);
     if (value.length === 1) value = value[0];
+  } else if (STRING_OPTIONS.has(key)) {
+    // TODO: refactor function to make this early return not
+    // stand out
+    return value;
   } else if (value.indexOf(':') > 0) {
```

Comment:
I think we also only want to take this path if the key is `readpreferencetags `

Reply:
also `authMechanismProperties`

Reply:
should we special case those two explicitly?

Reply:
I tried doing it pretty quickly, and it led to errors. I think we should instead make a larger ticket to deal with this, where we hard-code everything from https://github.com/mongodb/specifications/blob/master/source/uri-options/uri-options.rst, and specifically parse based off of that. For now, I'd like to just get this in to fix the ticket.

Reply:
OK SGTM